cache_to_disk/__init__.py

```diff
@@ -71,37 +182,51 @@ def delete_old_disk_caches():
     new_cache_metadata = deepcopy(cache_metadata)
     cache_changed = False
     for function_name, function_caches in cache_metadata.items():
-        if function_name != 'total_number_of_cache_to_disks':
-            to_keep = []
-            for function_cache in function_caches:
-                max_age_days = int(function_cache['max_age_days'])
-                file_name = disk_cache_dir + function_cache['file_name']
-                if file_exists(file_name):
-                    if get_age_of_file(file_name) > max_age_days:
-                        cache_changed = True
-                        os.remove(file_name)
-                    else:
-                        to_keep.append(function_cache)
-                else:
-                    cache_changed = True
-            if to_keep:
-                new_cache_metadata[function_name] = to_keep
+        if function_name == _TOTAL_NUMCACHE_KEY:
+            continue
+        to_keep = []
+        for function_cache in function_caches:
+            max_age_days = int(function_cache['max_age_days'])
+            file_name = join_path(DISK_CACHE_DIR, function_cache['file_name'])
+            if not file_exists(file_name):
+                cache_changed = True
+                continue
+            if not get_age_of_file(file_name) > max_age_days != UNLIMITED_CACHE_AGE:
```

Comment:
I think this is a bit more readable: 

if (get_age_of_file(file_name) <= max_age_days) or (max_age_days == UNLIMITED_CACHE_AGE)

Reply:
You're certainly correct. My IDE was whining about the "long" form and because it had an option to automatically transform it, I decided to go with that instead of littering yet another `#noqa` about. Seems good to me to stick with the way it was before, much clearer. I actually had to double check that the logic actually *worked* when the IDE recommended it- it makes sense, but it's a bit mindblowing at first glance. Reminds me of code a compiler would emit...